scipy/_lib/_util.py

```diff
@@ -1345,3 +1345,15 @@ def wrapper(*args, **kwargs):
 
         return wrapper
     return decorator
+
+
+def np_vecdot(x1, x2, /, *, axis=-1):
+    # `np.vecdot` has advantages (e.g. see gh-22462), so let's use it when
+    # available. As functions are translated to Array API, this can be replaced
+    # with `array-api-strict` NumPy's vecdot.
```

Comment:
Optional nit but I think this is clearer (if this is what you mean).

```suggestion
    # with `xp.vecdot`.
```

Reply:
I wasn't thinking. I think I meant array API compat vecdot. But yes, `xp.vecdot` would work.